src/bitshift.rs

```diff
@@ -0,0 +1,265 @@
+// Manually translated to Rust from FastLED's MIT licensed C++ code
+
+/// Steal brightness from brightness_src and give it to brightness_dst.
+/// After this function concludes the multiplication of brightness_dst and brightness_src will remain constant.
+pub(crate) fn brightness_bitshifter8(
```

Comment:
This seems like the sort of thing that would be better served as an external effects crate or similar. I don't think this crate is the right place for it. Does it only apply to apa102 over spi?

Reply:
Yes, this is specific to the APA102 protocol, so I think it belongs in this crate.